logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -26,4 +60,76 @@ internal class CallDataSource(
             callApi.connectToSFT(url = url, data = data)
         }
     }
+
+    override fun getAllCalls(): StateFlow<List<Call>> = MutableStateFlow(allCalls.value.calls.values.toList())
+
+    override fun getIncomingCalls(): Flow<List<Call>> = allCalls.map {
+        it.calls.values.filter { call ->
+            call.status in listOf(
+                CallStatus.INCOMING
+            )
+        }
+    }
+
+    override fun getOngoingCall(): Flow<List<Call>> = allCalls.map {
+        it.calls.values.filter { call -> call.status == CallStatus.ESTABLISHED }
+    }
+
+    override suspend fun sendCallingMessage(
+        conversationId: ConversationId,
+        userId: UserId,
+        clientId: ClientId,
+        data: String
+    ): Either<CoreFailure, Unit> {
+        val messageContent = MessageContent.Calling(data)
+        val date = Clock.System.now().toString()
+        val message = Message(uuid4().toString(), messageContent, conversationId, date, userId, clientId, Message.Status.SENT)
+        return messageSender.sendMessage(message)
+    }
```

Comment:
`MessageSender` is on the domain layer, above the repositories, together with the `UseCases` and `CallManager`.

This business logic of getting the correct dates and creating a message with a generated UUID, which is only being used in `OnHttpRequest` could be left there.

Not really worth moving this logic down to the repo risking some circular dependency in the future and whatnot.

TL;DR: I suggest inlining this code inside `OnHttpRequest`, and don't depend on `MessageSender` in any repository.